From b21a2ad56e7ff0f85a7685e795fd3ba366a020e8 Mon Sep 17 00:00:00 2001
From: Harini Katakam <harini.katakam@xilinx.com>
Date: Thu, 24 Sep 2020 13:16:52 +0530
Subject: [PATCH 1614/1851] dt-bindings: Add devicetree documentation for
 Xilinx PTP timer

commit 0d50bca6dae11d194e05c32eb9417b662f732fdd from
https://github.com/Xilinx/linux-xlnx.git

Add DT documentation for Xilinx 1588 PTP timer.

Signed-off-by: Harini Katakam <harini.katakam@xilinx.com>
Reviewed-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../devicetree/bindings/ptp/ptp-xilinx.yaml   | 34 +++++++++++++++++++
 1 file changed, 34 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/ptp/ptp-xilinx.yaml

diff --git a/Documentation/devicetree/bindings/ptp/ptp-xilinx.yaml b/Documentation/devicetree/bindings/ptp/ptp-xilinx.yaml
new file mode 100644
index 000000000000..d88e22695976
--- /dev/null
+++ b/Documentation/devicetree/bindings/ptp/ptp-xilinx.yaml
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: GPL-2.0-or-later OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/ptp/xilinx/ptp-xilinx.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Xilinx 1588 PTP timer-syncer
+
+maintainers:
+  - Harini Katakam <harini.katakam@xilinx.com>
+
+description: |
+  Xilinx 1588 PTP timer-syncer IP provides  precision timing functionality
+  along with capabiity for time and fine adjustment. This IP is provided
+  for use with Xilinx Ethernet MAC with HW timestamping capability.
+
+properties:
+  compatible:
+    const: xlnx,timer-syncer-1588-1.0
+
+  reg:
+    description: |
+      Offset and length of the register set for the device.
+
+required:
+  - compatible
+  - reg
+
+examples:
+  - |
+    ptp_timer@80000000 {
+        compatible = "xlnx,timer-syncer-1588-1.0";
+        reg = <0x0 0x80000000 0x0 0x10000>;
+    };
-- 
2.31.1

